src/checks/checkExtension.js

```diff
@@ -11,15 +11,16 @@ governing permissions and limitations under the License.
 */
 
 /**
- * Check that extension are listed in install.yml file
+ * Check that if the "extension" property is specified in the install.yml file it is an object containing the "serviceCode" key.
  *
  * @param {object} fileData
  * @returns {Promise<{message: string, status: string}>}
  */
-async function checkExtensions(fileData) {
-    if (!fileData.extension) {
+async function checkExtension(fileData) {
+    // "extension" is an optional property
+    if (fileData.extension && (fileData.extension.constructor.name !== 'Object' || !fileData.extension.serviceCode)) {
         return {
-            message: 'install.yml must specify extension points',
+            message: '"extension" must provide the "serviceCode" key.',
```

Comment:
Have you considered doing this validation using YAML Schema? For example, https://www.npmjs.com/package/yaml-schema-validator

Reply:
@roribio , I am planning to do so. Right now the structure of [install.yml](https://wiki.corp.adobe.com/display/AdobeCloudPlatform/App+Builder+Template+Install+Configuration) file is not finalized yet. I am planning to re-use the yaml schema once it is finalized and available.